src/services/cooperation.js

```diff
@@ -114,6 +115,37 @@ const cooperationService = {
       cooperation.availableQuizzes = removeArraysUniqueValues(cooperation.availableQuizzes, cooperation.finishedQuizzes)
       await cooperation.save()
     }
+  },
+
+  updateResourceCompletionStatus: async ({ id, currentUser, resourceId, completionStatus }) => {
+    const { id: currentUserId, role: currentUserRole } = currentUser
+
+    if (currentUserRole !== roles.STUDENT) {
+      throw createError(403, ROLE_REQUIRED_FOR_ACTION(roles.STUDENT))
+    }
+
+    const cooperation = await Cooperation.findById(id)
+    cooperationService._validateCooperationUser(cooperation, currentUserId)
+
+    let resourceIdExists = false
+
+    cooperation.sections.forEach((section) => {
+      section.resources.forEach((resource) => {
+        if (resource.resource.toString() === resourceId) {
+          resource.completionStatus = completionStatus
+          resourceIdExists = true
+        }
+      })
+    })
```

Comment:
I think you should use default `for...of` here. With for statement you can use `break` just to not perform additional interation after finding resource. Just little optimization